java/opendcs/src/main/java/decodes/gui/PropertiesEditPanel.java

```diff
@@ -95,6 +99,30 @@ public Component getTableCellRendererComponent(JTable table, Object value,
                 String pn = ((String) model.getValueAt(modelRow, 0)).toUpperCase();
                 PropertySpec ps = propHash.get(pn);
                 cr.setToolTipText(ps != null ? ps.getDescription() : "");
+                // Check if this property is required and highlight if missing
+                if (ps != null && model.isPropertyRequired(pn)) {
+                    boolean hasValue = value != null && !value.toString().trim().isEmpty();
+                    List<RequirementGroup> groups = model.getPropertyRequirementGroups(pn);
+                    boolean anyGroupSatisfied = true;
+                    
+                    // Check if at least one of the property's requirement groups is satisfied
+                    for (RequirementGroup group : groups) {
+                        if (!model.isRequirementGroupSatisfied(group.getGroupName())) {
+                            anyGroupSatisfied = false;
+                            break;
+                        }
+                    }
+                    
+                    // Highlight empty required fields that are part of unsatisfied groups
+                    if (!hasValue && !anyGroupSatisfied) {
```

Comment:
Perhaps put the above logic into a helper function that returns a Boolean pair?

This is a lot of business logic within a GUI handler.